server/src/WebHook.hs

```diff
@@ -126,7 +140,7 @@ postStatus uuid token domain commit s desc = flip catchError (logInfo_ . pack .
 
 -- | Run webhook jobs and post status to github
 runWebHook :: UUID -> Logger -> Config -> Commit -> IO ()
```

Comment:
I feel like that we are passing way too many parameters here and also in `postStatus` and `handleCompileResult`. A `ReaderT (Config, Commit) (LogT IO) ()` as the return type of all these functions could be better?